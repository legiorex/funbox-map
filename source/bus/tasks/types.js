export const types = {
    // Sync
    FILL_TASKS:           'FILL_TASKS',
    CREATE_TASK:          'CREATE_TASK',
    REMOVE_TASK:          'REMOVE_TASK',
    TOGGLE_FAVORITE_TASK: 'TOGGLE_FAVORITE_TASK',
    EDIT_MESSAGE_TASK:    'EDIT_MESSAGE_TASK',

    // Async
    FETCH_TASKS_ASYNC: 'FETCH_TASKS_ASYNC',
    CREATE_TASK_ASYNC: 'CREATE_TASK_ASYNC',
    REMOVE_TASK_ASYNC: 'REMOVE_TASK_ASYNC',
    UPDATE_TASK_ASYNC: 'UPDATE_TASK_ASYNC',
};
