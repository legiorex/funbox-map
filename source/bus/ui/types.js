export const types = {
    START_FETCHING:      "START_FETCHING",
    STOP_FETCHING:       "STOP_FETCHING",
    EMIT_ERROR:          "EMIT_ERROR",
    INPUT_TASK:          "INPUT_TASK",
    CLEAR_TASK:          "CLEAR_TASK",
    CHECK_ALL_TASKS:     "CHECK_ALL_TASKS",
    START_EDITING_TASK:  "START_EDITING_TASK",
    FINISH_EDITING_TASK: "FINISH_EDITING_TASK",
    CURRENT_INPUT:       "CURRENT_INPUT",
    NEW_MESSAGE_TASK:    "NEW_MESSAGE_TASK",
    PREV_MESSAGE:        'PREV_MESSAGE',
};
