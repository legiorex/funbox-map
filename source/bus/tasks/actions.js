// Types
import { types } from "./types";

// Instruments
import { api } from '../../REST';

export const tasksActions = {

    // Sync

    fillTasks: (tasks) => {
        return {
            type:    types.FILL_TASKS,
            payload: tasks,
        };
    },
    createTask: (task) => {
        return {
            type:    types.CREATE_TASK,
            payload: task,
        };
    },

    // Async

    fetchTasksAsync: () => {
        return {
            type: types.FETCH_TASKS_ASYNC,
        };

    },
    createTaskAsync: (newTaskMessage) => {
        return {
            type:    types.CREATE_TASK_ASYNC,
            payload: newTaskMessage,
        };
    },

};
