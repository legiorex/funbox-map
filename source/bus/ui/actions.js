// Types
import { types } from './types';

export const uiActions = {
    startFetching: () => {
        return {
            type: types.START_FETCHING,
        };
    },

    stopFetching: () => {
        return {
            type: types.STOP_FETCHING,
        };
    },
    inputTask: (value) => {

        return {
            type:    types.INPUT_TASK,
            payload: value,
        };
    },
    clearTask: () => {
        return {
            type: types.CLEAR_TASK,
        };
    },

    checkedAllTasks: (tasks) => {

        return {
            type:    types.CHECK_ALL_TASKS,
            payload: tasks,
        };
    },

    editTaskMessage: (newMessage) => {
        return {
            type:    types.NEW_MESSAGE_TASK,
            payload: newMessage,
        };
    },
    
    startEditingTask: (idTask) => {
        return {
            type:    types.START_EDITING_TASK,
            payload: idTask,
        };
    },
    finishEditingTask: () => {
        return {
            type: types.FINISH_EDITING_TASK,
        };
    },

    prevMessage: (message) => {
        return {
            type:    types.PREV_MESSAGE,
            payload: message,
        };
    },

    emitError: (error, meta = null) => {
        return {
            type:    types.EMIT_ERROR,
            payload: error,
            error:   true,
            meta,
        };
    },
};
