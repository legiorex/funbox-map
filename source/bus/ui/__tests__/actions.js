// Actions
import { uiActions } from "../actions";

// Types
import { types } from '../types';

describe('ui actions', () => {
    test('start fetching tasks', () => {
        expect(uiActions.startFetching()).toEqual({
            type: types.START_FETCHING,
        });
    });
    test('stop fetching tasks', () => {
        expect(uiActions.stopFetching()).toEqual({
            type: types.STOP_FETCHING,
        });
    });

    test('inputTask', () => {
        expect(uiActions.inputTask(__.message)).toEqual({
            type:    types.INPUT_TASK,
            payload: __.message,
        });
    });

    test('clear task', () => {
        expect(uiActions.clearTask()).toEqual({
            type: types.CLEAR_TASK,
        });
    });

    test('checkedAllTasks', () => {
        expect(uiActions.checkedAllTasks(__.tasks)).toEqual({
            type:    types.CHECK_ALL_TASKS,
            payload: __.tasks,
        });
    });

    test('editTaskMessage', () => {
        expect(uiActions.editTaskMessage(__.message)).toEqual({
            type:    types.NEW_MESSAGE_TASK,
            payload: __.message,
        });
    });

    test('startEditingTask', () => {
        expect(uiActions.startEditingTask(__.taskID)).toEqual({
            type:    types.START_EDITING_TASK,
            payload: __.taskID,
        });
    });

    test('finishEditingTask', () => {
        expect(uiActions.finishEditingTask()).toEqual({
            type: types.FINISH_EDITING_TASK,
        });
    });

    test('prevMessage', () => {
        expect(uiActions.prevMessage(__.message)).toEqual({
            type:    types.PREV_MESSAGE,
            payload: __.message,
        });
    });
    test('emitError', () => {
        expect(uiActions.emitError(__.error, null)).toMatchSnapshot();
    });
});
