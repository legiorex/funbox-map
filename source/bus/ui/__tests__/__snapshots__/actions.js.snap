// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions emitError 1`] = `
Object {
  "error": true,
  "meta": Object {},
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;
